lib/libvfio-user.c

```
@@ -1030,6 +1035,12 @@ handle_dma_device_feature_get(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
     struct vfio_user_device_feature_dma_logging_report *rep =
         (void *)req->data;
 
+    dma_controller_t *dma = vfu_ctx->dma;
+
+    if (dma == NULL) {
+        vfu_log(vfu_ctx, LOG_ERR, "DMA not enabled for DMA device feature");
```

Don't we need to return an error here?

---

Yes, whoops!